Added Docker development Environment.

xRahul opened the pull request:
- Data will be persisted for both mongo and mysql.
- Used docker compose to link web, mongo and mysql server

xRahul committed:
Added Docker development Environment.

Data can be persisted for both mongo and mysql

xRahul commented on public/install.php:
```diff
@@ -24,7 +24,7 @@
         \'DEFAULT_ACTION\' => \'index\',
 
         \'DB_TYPE\' => "' . $db_type . '",
-        \'DB_HOST\' => \'127.0.0.1\',
```
This has been changed so that docker environment can make use of db.
If you're developing in local, you'd only need to add this entry in `/etc/hosts`-
```
127.0.0.1    mysqlhost
```

xRahul commented on public/install.php:
```diff
@@ -63,7 +63,7 @@
                                   `email` varchar(64) COLLATE utf8_unicode_ci NOT NULL,
                                   `username` varchar(32) COLLATE utf8_unicode_ci NOT NULL,
                                   `phone` varchar(32) COLLATE utf8_unicode_ci NOT NULL,
-                                  `password` varchar(128) COLLATE utf8_unicode NOT NULL,
```
This was throwing error in the install script.

imujjwal96 replied:
i will rectify this error in the next commit.

xRahul replied:
I have already made the fix and added docker env for easier development.

imujjwal96 commented:
I will have to keep this PR on hold because i need to understand the logistics of Docker and for that i'll need some time.

xRahul commented:
@imujjwal96 sure! let me know if you need any help

imujjwal96 commented:
Cheers @xRahul. I definitely will do.

xRahul commented:
Hey @imujjwal96, Did you have a look at the docker?

google-labs-jules[bot] committed:
Modernize codebase with PHP 8.2, Namespacing, Tests, and CI/CD (#2)

* Modernize codebase to PHP 8.2, add PSR-4 namespacing, tests, and CI/CD pipelines

- Updated Docker environment to PHP 8.2 and Docker Compose v3.8.
- Refactored application to use PSR-4 namespaces (App\Core, App\Controllers, App\Models).
- Implemented environment variable-based configuration using phpdotenv.
- Modernized database connections (PDO MySQL, MongoDB).
- Added PHPUnit tests (Unit and Integration) and configured CI/CD with GitHub Actions.
- Removed insecure install.php and updated documentation.

* Fix MongoDB version mismatch in CI

- Downgraded `mongodb/mongodb` to `1.15.*` in `composer.json` to ensure compatibility with available `ext-mongodb` versions in standard CI environments.
- Updated `composer.lock`.

* Downgrade mongodb/mongodb to 1.12.* for broader extension compatibility

- Downgraded `mongodb/mongodb` from `1.15.*` to `1.12.*` in `composer.json`.
- This version requires `ext-mongodb ^1.13.0`, which is more likely to be available in restricted CI environments while still supporting `countDocuments()`.
- Updated `composer.lock`.

* Upgrade mongodb/mongodb to ^2.0 for ext-mongodb 2.x compatibility

- Upgraded `mongodb/mongodb` dependency to `^2.0` in `composer.json` to support `ext-mongodb` version 2.1.4 present in the CI environment.
- Updated `composer.lock`.
- Verified tests pass with the new library version.

---------

Co-authored-by: google-labs-jules[bot] <161369871+google-labs-jules[bot]@users.noreply.github.com>
